types/src/transaction/use_case.rs

```diff
@@ -81,16 +96,301 @@ impl UseCaseAwareTransaction for SignatureVerifiedTransaction {
         let payload =
             payload.expect("No payload found for SignatureVerifiedTransaction in parse_use_case");
 
-        match payload.executable_ref() {
-            Ok(TransactionExecutableRef::EntryFunction(entry_fun)) => {
-                let module_id = entry_fun.module();
+        let maybe_entry_func = match payload {
```

Comment:
I don't like the duplicate code, we should have a function taking the &Payload and return to unify the two cases in the file